src/Paket.Core/InstallModel.fs

```diff
@@ -115,171 +124,459 @@ type AnalyzerLib =
+            |> Option.map (fun (rid, l, _) -> { Path = l; File = p; Runtime = rid }))
+        |> Option.orElseWith (fun _ ->
+            (trySscanf "lib/%A{noSeperator}" p.PathWithinPackage : string option)
+            |> Option.map (fun (_) -> { Path = Tfm.Empty; File = p; Runtime = RuntimeIdentifier.Any }))
+
+    let getCompileLibAssembly (p:UnparsedPackageFile) =
+        // %s because 'native' uses subfolders...
+        (trySscanf "lib/%A{tfm}/%s" p.PathWithinPackage : (Tfm * string) option)
+        |> Option.map (fun (l,path) ->
+            if l.Name = "native" && l.Platforms = [ FrameworkIdentifier.Native(NoBuildMode,NoPlatform) ] then
+                // We need some special logic to detect the platform
+                let path = path.ToLowerInvariant()
+                let newPlatform =
+                    if path.Contains "/x86/" then Win32 else
+                    if path.Contains "/arm/" then Arm else
+                    if path.Contains "/x64/" then X64 else
+                    if path.Contains "/address-model-32" then UnknownPlatform "address-model-32" else
+                    if path.Contains "/address-model-64" then UnknownPlatform "address-model-64" else
+                    NoPlatform
+                let newBuildMode =
+                    if path.Contains "/release/" then Release else
+                    if path.Contains "/debug/" then Debug else
+                    NoBuildMode
+                { Path = { l with Platforms = [ FrameworkIdentifier.Native(newBuildMode,newPlatform) ]}; File = p; Runtime = RuntimeIdentifier.Any }
+            else
+            { Path = l; File = p; Runtime = RuntimeIdentifier.Any })
+        |> Option.orElseWith (fun _ ->
+            (trySscanf "lib/%A{noSeperator}" p.PathWithinPackage : string option)
+            |> Option.map (fun (_) -> { Path = Tfm.Empty; File = p; Runtime = RuntimeIdentifier.Any }))
+
+    let getRuntimeLibrary (p:UnparsedPackageFile) =
+        (trySscanf "runtimes/%A{rid}/nativeassets/%A{tfm}/%A{noSeperator}" p.PathWithinPackage : (Rid * Tfm * string) option)
+        |> Option.map (fun (rid, l,_) -> { Path = l; File = p; Runtime = rid })
+        |> Option.orElseWith (fun _ ->
+            (trySscanf "runtimes/%A{rid}/native/%A{noSeperator}" p.PathWithinPackage : (Rid * string) option)
+            |> Option.map (fun (rid, _) -> { Path = Tfm.Empty; File = p; Runtime = rid }))
+
+    let getMsbuildFile (p:UnparsedPackageFile) =
+        (trySscanf "build/%A{tfm}/%A{noSeperator}" p.PathWithinPackage : (Tfm * string) option)
+        |> Option.map (fun (l,_) -> { Path = l; File = p; Runtime = RuntimeIdentifier.Any })
+        |> Option.orElseWith (fun _ ->
+            (trySscanf "build/%A{noSeperator}" p.PathWithinPackage : string option)
+            |> Option.map (fun (_) -> { Path = Tfm.Empty; File = p; Runtime = RuntimeIdentifier.Any }))
```

Comment:
I think this part is now quite readable. I'm open for suggestions though. 
 - `trySscanf` is not completely type-safe (therefore the type annotations)
 - The `%A{scanner}` "extension" was invented by me to solve this problem